test/qa/test_meshnet.py

```diff
@@ -88,31 +89,211 @@ def test_mesh_removed_machine_by_other():
+        if not local:
+            continue
+
+        lan_allow_rule = f"-A FORWARD -s {peer_ip}/32 -d {lan} -m comment --comment nordvpn-exitnode-transient -j ACCEPT"
+        lan_allow_rule_index = rules.find(lan_allow_rule)
+
+        if lan_allow_rule not in rules:
+            return (False, f"LAN allow rule for subnet {lan} not found\nrules:\n{rules}")
+
+        if lan_allow_rule_index > lan_drop_rule_index:
+            return (False, f"LAN allow rule is added after LAN drop rule\nrules:\n{rules}")
+
+
+    return (True, "")
+
+
+def set_permissions(peer: str, routing: bool, local: bool, incoming: bool, fileshare: bool):
+    def bool_to_permission(permission: bool) -> str:
+        if permission:
+            return "allow"
+        return "deny"
+
+    # ignore any failures that might occur when permissions are already configured to the desired value
+    sh.nordvpn.mesh.peer.routing(bool_to_permission(routing), peer, _ok_code=(0, 1))
+    sh.nordvpn.mesh.peer.local(bool_to_permission(local), peer, _ok_code=(0, 1))
+    sh.nordvpn.mesh.peer.incoming(bool_to_permission(incoming), peer, _ok_code=(0, 1))
+    sh.nordvpn.mesh.peer.fileshare(bool_to_permission(fileshare), peer, _ok_code=(0, 1))
+
+
+@pytest.mark.parametrize("routing", [True, False])
+@pytest.mark.parametrize("local", [True, False])
+@pytest.mark.parametrize("incoming", [True, False])
+@pytest.mark.parametrize("fileshare", [True, False])
+def test_exitnode_permissions(routing: bool,
+                              local: bool,
+                              incoming: bool,
+                              fileshare: bool):
+    peer_ip = meshnet.get_this_device_ipv4(ssh_client.exec_command("nordvpn mesh peer list"))
+    set_permissions(peer_ip, routing, local, incoming, fileshare)
+
+
+    def is_filter_correct():
+        return validate_input_chain(peer_ip, routing, local, incoming, fileshare)
+
+    # We only need to poll the first check, after the first check succeeded, all of the other checks should be ready.
+    for (result, message) in lib.poll(is_filter_correct):
+        if result:
+            break
+    assert result, message
+
+    (result, message) = validate_forward_chain(peer_ip, routing, local, incoming, fileshare)
+    assert result, message
+
+    rules = sh.sudo.iptables("-S", "POSTROUTING", "-t", "nat")
+
+    if routing:
+        assert f"-A POSTROUTING -s {peer_ip}/32 -o eth0 -m comment --comment nordvpn -j MASQUERADE" in rules
+    else:
+        assert f"-A POSTROUTING -s {peer_ip}/32 -o eth0 -m comment --comment nordvpn -j MASQUERADE" not in rules
+
+
+@pytest.mark.parametrize("lan_discovery", [True, False])
+@pytest.mark.parametrize("local", [True, False])
+def test_lan_discovery_exitnode(lan_discovery: bool, local: bool):
+    peer_ip = meshnet.get_this_device_ipv4(ssh_client.exec_command("nordvpn mesh peer list"))
+    set_permissions(peer_ip, True, local, True, True)
+
+    lan_discovery_value = "on" if lan_discovery else "off"
```

Comment:
Wouldn't need this if you just did `@pytest.mark.parametrize("lan_discovery", ["on", "off"])`

Reply:
Yeah, but we also need it as a bolean later on, for parametrization. So it's the same difference really.

Reply:
Right, missed it, specifically didn't write this comment above for the reason you mentioned. Although technically some lines could still be saved if you used my parametrization and when you needed a boolean just did `lan_discovery == "on"` :D